src/shared/loop-util.c

```diff
@@ -399,6 +399,26 @@ static int attach_empty_file(int loop, int nr) {
         return 0;
 }
 
+/* Duplicate a file descriptor without O_DIRECT for unaligned probing */
+static int loop_device_fd_dup_for_dissect(int loop_fd) {
```

Comment:
BTW, I am not familiar with the difference, but is this possible to use `fd_reopen()`?